bundles/tools.vitruv.change.propagation/src/tools/vitruv/change/propagation/impl/DefaultChangeRecordingModelRepository.java

```diff
@@ -170,6 +172,8 @@ public Iterable<TransactionalChange> recordChanges(Runnable changeApplicator) {
 
 	@Override
 	public VitruviusChange applyChange(VitruviusChange change) {
+		// Ensure all resources for changed elements are created/loaded and change monitors are initialized
+		FluentIterable.from(change.getChangedURIs()).forEach(uri -> getModelResource(uri));
```

Comment:
The method name here is a bit unfortunate as we actually rely on the side-effect of registering the model for recording in `getModelResource`. Maybe we can refactor this behaviour into a better named method? This would also remove the need for the explaining comment.

Reply:
True. We can actually call `getCreateOrLoadModel`, as that's the realization method that is called by `getModelResource`. This makes sense, because the need to call that methods is given because of the realization of the `getModelResource` API method in this specific class. We cannot change the name of the `getModelResource` method, as it is an API method and the API abstracts from creating/loading/retrieving etc.
In addition, calling `getCreateOrLoadModel` would make the comment obsolete, right, @JanWittler?